session/session.go

```diff
@@ -1276,6 +1281,29 @@ func (s *session) ExecuteStmt(ctx context.Context, stmtNode ast.StmtNode) (sqlex
 	return recordSet, nil
 }
 
+func (s *session) validateStatementReadOnlyInStaleness(stmtNode ast.StmtNode) error {
+	vars := s.GetSessionVars()
+	if !vars.TxnCtx.IsStaleness {
+		return nil
+	}
+	errMsg := "only support read only statement during time-bounded read only transaction"
+	switch stmtNode.(type) {
+	case *ast.SplitRegionStmt:
+		return nil
+	case *ast.SelectStmt, *ast.ExecuteStmt, *ast.ExplainStmt, *ast.ShowStmt, *ast.SetOprStmt:
```

Comment:
`ExecuteStmt` is not covered by `IsReadOnly`.
`DoStmt` should also be allowed to execute.

Reply:
I think `ExecuteStmt` is covered by `planner.IsReadOnly`. We can also see it from the testcase `execute insert statement`

Reply:
`doStmt` is allowed, the testcases is updated.